bin/main.rb

```diff
@@ -1,2 +1,124 @@
-#!/usr/bin/env ruby
-puts 'Hello World'
+# !/usr/bin/env ruby
+# Implement TicTacToe Game Interface
+
+class TicTacToe
+  def initialize
+    @position = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @count = 0
+    puts 'Enter First Player name: '
+    @name1 = gets.chomp
+    puts 'Enter Second Player name: '
+    @name2 = gets.chomp
+    @player1 = 'X'
+    @player2 = 'O'
+    puts "#{@name1} is #{@player1}"
+    puts "#{@name2} is #{@player2}"
+  end
+
+  def draw_board
+    if @count == 9
+      @position = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+      @count = 0
+    end
+    puts "|  #{@position[6]}  |  #{@position[7]}  |  #{@position[8]} |"
+    puts '------------------'
+    puts "|  #{@position[3]}  |  #{@position[4]}  |  #{@position[5]} |"
+    puts '------------------'
+    puts "|  #{@position[0]}  |  #{@position[1]}  |  #{@position[2]} |"
+    puts '------------------'
+  end
+
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/AbcSize
+
+  def player_turn
+    draw_board
+    loop do
+      player = if @count.even?
+                 @name1
+               else
+                 @name2
+               end
+      puts "#{player} Make a move? Choose between 1-9"
+      move = gets.chomp.strip.to_i
+      update_board(move)
+      @count += 1
+      break if win?(player)
+      break if draw?
+    end
+  end
+
+  def update_board(move)
+    if move >= 1 && move <= 9
+      if @position[move - 1] != @player1 && @position[move - 1] != @player2 # If not already taken
+        if @count.even?
+          @position[move - 1] = @player1
+        elsif @count.odd?
+          @position[move - 1] = @player2
+        end
+      else
+        puts 'Already taken, Make another move'
+        @count -= 1
+      end
+    else
+      puts 'Invalid move'
+      @count -= 1
+    end
+    draw_board
+  end
+
+  def win?(player)
+    if @count.odd?
+      player = @player1
+      name = @name1
+    else
+      player = @player2
+      name = @name2
+    end
```

Comment:
This function is redundant since you are already submitting player variable witch is already been tested before for odd or even. I'm confident you can simplify this operation.

Reply:
I don't think so. This is not going to only switch but also allow to compare the winning move with just one argument "player" and also print one name which is whoever made the last move.